decode.go

```diff
@@ -153,6 +153,8 @@ func unmarshalReflect(av *dynamodb.AttributeValue, rv reflect.Value) error {
 		case rv.Type().Elem().Kind() == reflect.Bool:
 			truthy = reflect.ValueOf(true)
 		case rv.Type().Elem() == emptyStructType:
+			fallthrough
+		case rv.Type().Elem().Kind() == reflect.Struct && rv.Type().Elem().NumField() == 0:
 			truthy = reflect.ValueOf(struct{}{})
```

Comment:
This might be better implemented as `truthy = reflect.Zero(rv.Type().Elem())`. Thoughts?

Reply:
BTW I think both ways are fine, but your original way makes it a little more clear what exactly we're doing, so I left it as is 👍